### PR TITLE
✨ Add support for Pydantic v2

### PR DESCRIPTION
**Pull request by tiangolo**
✨ Add support for Pydantic v2

### CODE REVIEW

**Comment by tiangolo** [comment]
📝 Docs preview for commit b75e2fa944ae15f98dc154f4db719f08c0cd8a66 at: https://64a6a287b9883f18303a7b87--fastapi.netlify.app

**Comment by tiangolo** [comment]
📝 Docs preview for commit 792b032b771c64685258aadd67ee9e09fca0bf57 at: https://64a81e365713d115c6ff9969--fastapi.netlify.app

**Comment by tiangolo** [comment]
📝 Docs preview for commit 2d22f695ab5acd438b4311f68d79dde561d018f3 at: https://64a820e7539478007816ad4c--fastapi.netlify.app

**Comment by tiangolo** [comment]
📝 Docs preview for commit e676874f76ff728af2fe60eebf50246d630e6caa at: https://64a8450f16305408f0e1ea7e--fastapi.netlify.app

File: tests/test_jsonable_encoder.py [hunk 1]
```diff
@@ -227,19 +236,72 @@ class MyEnum(Enum):
     assert encoded_instance == custom_enum_encoder(instance)
 
 
-def test_encode_model_with_path(model_with_path):
-    if isinstance(model_with_path.path, PureWindowsPath):
-        expected = "\\foo\\bar"
-    else:
-        expected = "/foo/bar"
-    assert jsonable_encoder(model_with_path) == {"path": expected}
+def test_encode_model_with_pure_path():
```
Comment (iudeen) [inline tests/test_jsonable_encoder.py:239]: This is failing in Windows

Comment (iudeen) [reply tests/test_jsonable_encoder.py:239]: https://github.com/tiangolo/fastapi/pull/9840 can fix this

**Pull request merged**